projects/igniteui-angular/src/lib/grids/grid/grid.component.html

```diff
@@ -117,7 +122,7 @@
             [igxForItemSize]="hasColumnLayouts ? rowHeight * multiRowLayoutRowSize + 1 : renderedRowHeight"
             #verticalScrollContainer (onChunkPreload)="dataLoading($event)">
             <ng-template #record_template>
-                <igx-grid-row [gridID]="id" [index]="rowIndex" [rowData]="rowData" #row>
+                <igx-grid-row [gridID]="id" [index]="getGridRowIndex(rowData, rowIndex)" [rowData]="rowData" #row>
```

Comment:
Oh, this can have way more impact than you'd imagine (see tests).
Keep the different index inside the row template outlets context only!